BinaryObjectScanner/Protection/AlphaROM.cs

```diff
@@ -84,5 +88,50 @@ public class AlphaROM : IExecutableCheck<PortableExecutable>
 
             return null;
         }
+        
+        public string? CheckISO(string file, ISO9660 iso, bool includeDebug)
+        {
+            // Checks can be made even easier once UDF support exists, as most (although not all, some early discs like
+            // redump ID 124111 have no UDF partition) discs have "Settec" slathered over every field UDF lets them.
+
+            if (iso.VolumeDescriptorSet[0] is not PrimaryVolumeDescriptor pvd)
+                return null;
+            
+            
+            // Alpharom disc check #1: disc has varying (but observed to at least always be larger than 14) length 
+            // string made up of numbers and capital letters.
+            // TODO: triple-check that length is never below 14
+            int offset = 0;
+            var applicationIdentifierString = pvd.ApplicationIdentifier.ReadNullTerminatedAnsiString(ref offset)?.Trim();
+            if (applicationIdentifierString == null || applicationIdentifierString.Length < 14)
+                return null;
+
+            if (!Regex.IsMatch(applicationIdentifierString, "^[A-Z0-9]*$"))
+                return null;
+            
+            offset = 0;
```

Comment:
This offset reset doesn't seem to be used?

Reply:
I'm not saying the variable itself is an issue. I'm saying that setting the value back to 0 serves no obvious purpose in the following code in most places. You'd only need to do that if you then had to read from the starting offset again. It isn't like a stream position where this value is stateful to the actual array nor is it like a `byte*`

Reply:
It was unclear in some cases. I saw some that explicitly set it to a _new_ value and not use `0` again.